chain/consensus/mir/pool/pool.go

```diff
@@ -2,29 +2,40 @@
 package pool
 
 import (
+	"context"
+	"fmt"
+
 	"github.com/filecoin-project/mir/pkg/pb/requestpb"
 	requestpbtypes "github.com/filecoin-project/mir/pkg/pb/requestpb/types"
 )
 
 type Fetcher struct {
 	ReadyForTxsChan chan chan []*requestpb.Request
+	ctx             context.Context
 }
 
-func NewFetcher(ch chan chan []*requestpb.Request) *Fetcher {
+func NewFetcher(ctx context.Context, ch chan chan []*requestpb.Request) *Fetcher {
 	return &Fetcher{
+		ctx:             ctx,
```

Comment:
As discussed before, I find the Fetcher way more complicated than it needs to be, now even storing a context. I suggest creating an issue as a reminder to simplify (by directly calling the Eudico mempool's API) the code. It's not something critical to do now though.